internal/consensus/state.go

```diff
@@ -962,9 +962,15 @@ func (cs *State) handleMsg(mi msgInfo) {
 
 	case *CommitMessage:
 		cs.Logger.Info("XX received cmt message")
-		panic("RECEIVED COMMIT MESSAGE")
 		// TODO Commit has been validated Validate Basic when unmarshalling, but need to validate the the commit itself
-		//cs.AddCommit(aggCommit)
+		added, err := cs.AddCommit(msg.Commit, peerID)
+		if added {
+			cs.statsMsgQueue <- mi
```

Comment:
Can we handle stats of added commits? If not sure, leave a // XXX comment saying it will be done later